webdriver/tests/bidi/script/call_function/exception_details.py

```diff
@@ -131,9 +131,17 @@ async def test_invalid_function(bidi_session, top_context):
         ("new WeakMap()", {"type": "weakmap", },),
         ("new WeakSet()", {"type": "weakset", },),
         ("new Error('SOME_ERROR_TEXT')", {"type": "error"},),
-        # TODO(sadym): add `iterator` test.
-        # TODO(sadym): add `generator` test.
-        # TODO(sadym): add `proxy` test.
+        ("([1, 2][Symbol.iterator]())", {
+            "type": "iterator",
+            "value": [],
```

Comment:
The iterator doesn't have a `value` field. Doesn't that fail for Chrome?

Reply:
Interestingly, the test passes in both cases (with and without `value`.). Removed.

Reply:
I don't see that this type is marked as extensible. So this should basically fail the test, right?

Reply:
It doesn't. Anyway, the test is correct now, right? And it doesn't use AnyExtending either. The reason why it passes in chrome in both scenarios is out of scope of this PR. 

I'll just note that the generator test is failing for us as it outputs type: function but it should be fixed on our end.

Reply:
Sure, but note that the current implementation of recursive compare ignores fields that haven't explicitly specified to check. As such any additional field is accepted even for types that are not extensible. I can remember that an alternative solution has been started but that was never completed.

Reply:
> I can remember that an alternative solution has been started but that was never completed.

You're probably referring to https://github.com/web-platform-tests/wpt/pull/38823. I am hoping to restart it at some point, but the current CI state (that requires admin force-merge for every PR) makes it infeasible to iterate quickly in order to submit those changes. In fact, let me open a separate bug to address this.

Reply:
Filed https://github.com/web-platform-tests/wpt/issues/40860